src/Admin/SiteHealth.php

```diff
@@ -0,0 +1,294 @@
+					'fields'      => [
+						'amp_mode_enabled'        => [
+							'label'   => 'AMP mode enabled',
+							'value'   => AMP_Theme_Support::get_support_mode(),
+							'private' => false,
+						],
+						'amp_experiences_enabled' => [
+							'label'   => esc_html__( 'AMP experiences enabled', 'amp' ),
+							'value'   => $this->get_experiences_enabled(),
+							'private' => false,
+						],
+						'amp_templates_enabled'   => [
+							'label'   => esc_html__( 'Templates enabled', 'amp' ),
+							'value'   => $this->get_supported_templates(),
+							'private' => false,
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Gets the AMP experiences that are enabled.
+	 *
+	 * @return string The experiences, in a comma-separated string.
+	 */
+	public function get_experiences_enabled() {
+		$experiences = AMP_Options_Manager::get_option( 'experiences' );
+		if ( empty( $experiences ) ) {
+			return esc_html__( 'No experience enabled', 'amp' );
+		}
+
+		return implode( ', ', $experiences );
+	}
+
+	/**
+	 * Gets the templates that support AMP.
+	 *
+	 * @return string The supported template(s), in a comma-separated string.
+	 */
+	public function get_supported_templates() {
+		$possible_post_types = AMP_Options_Manager::get_option( 'supported_post_types' );
+
+		// Get the supported content types, like 'post'.
+		$supported_templates = array_filter(
+			AMP_Post_Type_Support::get_eligible_post_types(),
+			static function( $template ) use ( $possible_post_types ) {
+				$post_type = get_post_type_object( $template );
+				return (
+					post_type_supports( $post_type->name, AMP_Post_Type_Support::SLUG )
+					||
+					( ! AMP_Options_Manager::is_website_experience_enabled() && in_array( $post_type->name, $possible_post_types, true ) )
+				);
+			}
+		);
+
+		// Add the supported templates, like 'is_author', if not in 'Reader' mode.
+		if ( AMP_Theme_Support::READER_MODE_SLUG !== AMP_Theme_Support::get_support_mode() ) {
+			$supported_templates = array_merge(
+				$supported_templates,
+				array_keys(
+					array_filter(
+						AMP_Theme_Support::get_supportable_templates(),
+						static function( $option ) {
+							return (
+								( empty( $option['immutable'] ) && ! empty( $option['user_supported'] ) )
+								||
+								! empty( $option['supported'] )
+							);
+						}
+					)
+				)
+			);
+		}
+
+		if ( empty( $supported_templates ) ) {
+			return esc_html__( 'No template supported', 'amp' );
+		}
+
+		return implode( ', ', $supported_templates );
+	}
+
+	/**
+	 * Adds a PHP extension to those that Core depends on.
+	 *
+	 * @param array $extensions The existing extensions from Core.
+	 * @return array The extensions, with one more added.
+	 */
+	public function add_extension( $extensions ) {
+		return array_merge(
+			$extensions,
+			[
+				'spl' => [
+					'extension' => 'spl',
+					'function'  => 'spl_autoload_register',
+					'required'  => true,
+				],
+			]
+		);
```

Comment:
Since `spl` is required, the AMP plugin won't even initialize without it, right? 

https://github.com/ampproject/amp-wp/blob/e6f9dc3587954e40c0c21bc66ccfa88f99aaa1b3/amp.php#L72-L74

So is `spl` relevant here? The test would never fail.

What is relevant are tests for the `suggest` packages:

https://github.com/ampproject/amp-wp/blob/e6f9dc3587954e40c0c21bc66ccfa88f99aaa1b3/composer.json#L29-L34

Reply:
Thanks, that's a good point. Tomorrow might be the earliest I could make that change, if that's alright.